osu.Game.Rulesets.Osu/Beatmaps/OsuBeatmapConverter.cs

```diff
@@ -43,7 +43,8 @@ protected override IEnumerable<OsuHitObject> ConvertHitObject(HitObject original
                     Position = positionData?.Position ?? Vector2.Zero,
                     NewCombo = comboData?.NewCombo ?? false,
                     ComboOffset = comboData?.ComboOffset ?? 0,
-                    LegacyLastTickOffset = legacyOffset?.LegacyLastTickOffset
+                    LegacyLastTickOffset = legacyOffset?.LegacyLastTickOffset,
+                    TickDistanceMultiplier = beatmap.BeatmapInfo.BeatmapVersion < 8 ? 1f / beatmap.ControlPointInfo.DifficultyPointAt(original.StartTime).SpeedMultiplier : 1
```

Comment:
```suggestion
                    // prior to v8, speed multipliers affect how many ticks are generated.
                    TickDistanceMultiplier = beatmap.BeatmapInfo.BeatmapVersion < 8 ? 1f / beatmap.ControlPointInfo.DifficultyPointAt(original.StartTime).SpeedMultiplier : 1
```

Reply:
Want to try and add inline comments for all the special version-specific cases (makes it a lot easier to figure what is going on, based on how hard it is to understand some of stable's code in these areas).

Reply:
Applied, but changed to `don't affect`.

Reply:
```
Prior to v8, `TickDistance` was not multiplied by the control point speed multiplier, resulting in the same number of ticks being present in the slider regardless of the speed of the slider. This is fixed by eliminating the speed multiplier through the following division.
```

Sound better? Imo the single line comment is self-explanatory enough.

Reply:
Again, you're thinking duration. But this has _nothing_ to do with duration and only to do with physical distance.

Reply:
I get that. But the *use case* for this being changed is all to do with duration since it is related to how mappers used that settings. That's what I want to capture in the comment.